### ISSUE TITLE
`finally` doesn't rethrow the stacktrace

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#27484]
```julia
julia> corrupt_stacktrace() = error("destroy stacktrace")

julia> f() = error("expected")
f (generic function with 1 method)

julia> try
        f()
       finally
        try; corrupt_stacktrace(); end
       end
ERROR: 
Stacktrace:
 [1] error at ./error.jl:33 [inlined]
 [2] corrupt_stacktrace at ./REPL[8]:1 [inlined] # should have been f!
 [3] macro expansion at ./REPL[14]:4 [inlined]
 [4] top-level scope at ./<missing>:0
```

**Comment by JeffBezanson** [comment]
Duplicate of #19979?

**Comment by vtjnash** [comment]
No, those are all for try/catch. This one is reporting that try/finally is also broken, and that none of the workarounds or fixes mentioned there would be applicable.

**Comment by JeffBezanson** [comment]
I think @c42f 's proposal of a stack of exceptions would fix it; after the inner `try; corrupt... catch` the backtrace would be reset to the previous one.

**Comment by c42f** [comment]
Yes, I think an exception stack would fix this just as it fixes various other problems of nested exception handling.

**Comment by c42f** [comment]
Fixed in #28878
